--- tokyocabinet/table.c.0	2010-12-11 09:04:11.000000000 +0900
+++ tokyocabinet/table.c	2010-12-11 09:05:16.000000000 +0900
@@ -68,7 +68,7 @@
     }
     
     PyObject *key, *value;
-    int pos = 0;
+    Py_ssize_t pos = 0;
     const char *kstr, *vstr;
     TCMAP *map;
     
@@ -1255,7 +1255,7 @@
 Table_subscript(Table *self, PyObject *key)
 {
     char *kbuf;
-    int ksiz;
+    Py_ssize_t ksiz;
     TCMAP *cols;
     PyObject *value;
     
@@ -1292,7 +1292,7 @@
 {
     bool success;
     char *kbuf;
-    int ksiz;
+    Py_ssize_t ksiz;
     TCMAP *cols;
     
     if (!PyString_Check(key))
