--- setup.py.0	2010-11-07 20:00:35.874587352 +0900
+++ setup.py	2010-11-07 20:06:50.545857939 +0900
@@ -101,13 +101,9 @@
     # Load dependancies
     dependencies = []
     jars = []
-    for dependency,include in pom.download_dependencies(dependencies_path):
-        dependencies.append(dependency)
-        if True or include: #XXX: Add the jar (file only) to the inclusions list
-            jars.append(path.split(dependency)[1])
+    if not path.exists(dependencies_path):
+    	os.mkdir(dependencies_path)
     test.dependencies = test_dependencies = []
-    for jar in jars:
-        test_dependencies.append(path.join(dependencies_path,jar))
     # Configure setup
     packages = ['neo4j', 'neo4j.model',
                 'neo4j._backend', 'neo4j._hooks',
